relayer/provider/cosmos/tx.go

```diff
@@ -1178,6 +1178,37 @@ func (cc *CosmosProvider) MsgTimeoutOnClose(ctx context.Context, msgRecvPacket p
 	return NewCosmosMessage(msgTimeout), nil
 }
 
+func (cc *CosmosProvider) MsgUpdateClientHeader(latestHeader provider.IBCHeader, trustedHeight clienttypes.Height, trustedHeader provider.IBCHeader) (ibcexported.Header, error) {
+	trustedCosmosHeader, ok := trustedHeader.(CosmosIBCHeader)
+	if !ok {
+		return nil, fmt.Errorf("Unsupported IBC trusted header type. Must be CosmosIBCHeader: %v", trustedHeader)
```

Comment:
Error messages begin with a lowercase letter (unless referencing a proper noun).

It would be helpful to mention the type of the received value, with the %T verb.

Reply:
There is definitely a linter that can catch most of these... Offhand I can't remember exactly, but I think golangci-lint has one.

Reply:
linter catching these would be great, going between logging errors and constructing errors I end up doing this pretty often.

Reply:
done